libkbfs/bserver_remote.go

```diff
@@ -27,11 +26,18 @@ const (
 	BServerTokenExpireIn = 2 * 60 * 60 // 2 hours
 )
 
+type blockServerRemoteConfig interface {
+	diskBlockCacheGetter
+	codecGetter
+	signerGetter
+	currentInfoGetterGetter
+	logMaker
+}
+
 // BlockServerRemote implements the BlockServer interface and
 // represents a remote KBFS block server.
 type BlockServerRemote struct {
-	codec      kbfscodec.Codec
-	cig        currentInfoGetter
+	config     blockServerRemoteConfig
```

Comment:
Oh, forgot to mention. One reason of having a `fooConfig` interface instead of inlining the arguments is to be able to switch out the various fields at runtime, e.g. for tests. So if a type doesn't need that, then it doesn't need a config interface.

(Of course, that reason is itself not a very good one, since one could just provide a switchable implementation of a particular interface for tests instead of adding an extra level of indirection. But we're stuck with what we have (`Config`) unless we want to rewrite the world.)